src/utils/roles.ts

```diff
@@ -63,18 +72,10 @@ export async function addNewRole<T extends IRole>(
   id: string
 ) {
   // This function is triggered when a user changes their role, it adds the new role to the user
-  let role;
-  switch (model.modelName) {
-    case "class":
-      //TODO: Rewrite IRole to include name, and replace CODE with name
-      role = await model.findOne({ CODE: id });
-      break;
-    default:
-      role = await model.findOne({ NAME: id });
-  }
+  const role = await model.findOne({ ROLE_NAME: id });
```

Comment:
`ROLE_NAME` should be changed to `NAME`
`NAME` is the pk in the database used to match the interaction.option to the db entry. maybe the pk name should be changed to `ID` or something to reduce confusion.

Reply:
I agree. I swapped the variable in the function to be name, then added a note in documentation, we can open an issue to swap it to name or code